xclim/indices/_agro.py

```diff
@@ -263,12 +263,14 @@ def huglin_index(
         )
         k_aggregated = 1
     elif method.lower() == "jones":
-        day_length = aggregate_between_dates(
-            day_lengths(dates=tas.time, lat=lat, method="simple"),
-            start=start_date,
-            end=end_date,
-            op="sum",
-            freq=freq,
+        day_length = (
```

Comment:
Indeed, the same unoptimized behaviour will happen for this index. As it is the only one now using `aggregate_between_dates`, one option we have is to rewrite the index a single custom function. We might be able to achieve better performance then.

However, I think this idea of time-variable time bounds is still relevant. So I see two ways forward:

1. A  `select_time` that accepts variable and multidimensional bounds. We want to compute bounds for each period and then go back to the daily input and mask the dates outside those bounds. The mask may have spatial dimensions too.
2. An optimized `resample().map()`. By this I mean, that passing a generic function to the resampler would _not_ fragment the chunks to 1 per period.

Solution  2 is clearly outside the scope of xclim. This is work for developpers of `flox` and `xarray`, and I don't think it will be easy, if even possible.

Solution 1 will optimize the case where bounds must be dynamic, but the resampling function is simple (mean, sum, etc). I think this is the case for `effective_growing_degree_days`.

Reply:
Agreed that solution 2 sounds a bit above our pay grade. How much effort are we talking if you were to implement solution 1?

I'm wondering if there are any other indicators that could benefit from dynamic start and end dates. Maybe there are some statistically-informed indexes that could be fascinating down the line? I don't think we'd lose much from keeping it around and optimizing it a bit.

Reply:
Effort for 1 might be large. Dask and xarray have changed a lot since we wrote `aggregate_between_dates` so it might be easier, but for the moment, I don't think ``effective_growing_degree_days ` (as the single use case) is worth it...

Reply:
Sounds good. Just wanted to be sure. This PR is good to go as is!